### PR TITLE
Fix dropdown translations on search results

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by trasher** [comment]
Second commit is not directly related since this code is not used in core.

File: src/Search.php [hunk 1]
```diff
@@ -1156,7 +1156,16 @@ public static function explodeWithID($pattern, $subject)
      */
     public static function joinDropdownTranslations($alias, $table, $itemtype, $field)
     {
-        return SQLProvider::joinDropdownTranslations($alias, $table, $itemtype, $field);
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        Toolbox::deprecated();
+
+        return "LEFT JOIN " . $DB::quoteName('glpi_dropdowntranslations') . " AS " . $DB::quoteName('$alias') . "
```
Comment (cedric-anne) [inline src/Search.php:1164]:
```suggestion
        return "LEFT JOIN " . $DB::quoteName('glpi_dropdowntranslations') . " AS " . $DB::quoteName($alias) . "
```

File: src/Search.php [hunk 2]
```diff
@@ -1156,7 +1156,16 @@ public static function explodeWithID($pattern, $subject)
      */
```
Comment (cedric-anne) [inline src/Search.php:1156]:
```suggestion
     * @deprecated 11.0.0
     */
```

File: src/Search.php [hunk 3]
```diff
@@ -1156,7 +1156,16 @@ public static function explodeWithID($pattern, $subject)
...
     public static function joinDropdownTranslations($alias, $table, $itemtype, $field)
     {
-        return SQLProvider::joinDropdownTranslations($alias, $table, $itemtype, $field);
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        Toolbox::deprecated();
```
Comment (cedric-anne) [inline src/Search.php:1162]: An entry in the changelog should be added.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Search.php (hunk 3 above)
Comment (trasher) [reply src/Search.php:1163]: Done

**Review by cedric-anne — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Pull request merged**